pkg/placement/what-resolver.go

```diff
@@ -189,27 +180,28 @@ func (wrh WhatResolverClusterHandler) OnDelete(obj any) {
 }
 
 func (wr *whatResolver) enqueue(gk schema.GroupKind, objAny any) {
-	key, err := kcpcache.DeletionHandlingMetaClusterNamespaceKeyFunc(objAny)
+	key, err := upstreamcache.DeletionHandlingMetaNamespaceKeyFunc(objAny)
 	if err != nil {
 		wr.logger.Error(err, "Failed to extract object reference", "object", objAny)
 		return
 	}
-	cluster, namespace, name, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	namespace, name, err := upstreamcache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		wr.logger.Error(err, "Impossible! SplitMetaClusterNamespaceKey failed", "key", key)
 	}
 	if namespace != "" {
 		panic("Namespace must be empty here")
 	}
-	item := namespacedQueueItem{GK: gk, Cluster: cluster, NN: NewPair(NamespaceName(metav1.NamespaceNone), ObjectName(name))}
+	// Enqueuewith empty cluster. Set it later
```

Comment:
This is ambiguous in general. The right solution will involve `enqueue` being given the cluster name as an additional parameter.